dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -95,6 +106,42 @@ class CZipExtract :
 
         int OnWizardNext()
         {
+            if (m_hExtractionMutex != INVALID_HANDLE_VALUE)
+            {
+                WaitForSingleObject(m_hExtractionMutex, INFINITE);
+                if (m_bExtractionThreadOver)
+                {
+                    if (m_bExtractionThreadSuccess)
+                    {
+                        ReleaseMutex(m_hExtractionMutex);
+                        return 0;
```

Comment:
Should return `TRUE` (see the "return value" comment below)

Reply:
This is the only place where we want to go to the next page, so I'm not sure how we can continue the wizard if we return TRUE here, unless I'm missing something.
(The previous version returned 0 after the extraction. This does the same, but we have to reactivate the "Next" button and simulate a click on it, hence why we have this whole paragraph at the start on OnWizardNext)